src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/create/workflow/steps/PopulateNewPendingImportObjectsStep.java

```diff
@@ -0,0 +1,615 @@
+import org.breedinginsight.brapps.importer.services.processors.experiment.ExperimentUtilities;
+import org.breedinginsight.brapps.importer.services.processors.experiment.create.model.PendingData;
+import org.breedinginsight.brapps.importer.services.processors.experiment.create.model.PendingImportObjectData;
+import org.breedinginsight.brapps.importer.services.processors.experiment.create.model.ProcessContext;
+import org.breedinginsight.brapps.importer.services.processors.experiment.create.model.ProcessedPhenotypeData;
+import org.breedinginsight.brapps.importer.services.processors.experiment.services.ExperimentSeasonService;
+import org.breedinginsight.brapps.importer.services.processors.experiment.services.ExperimentValidateService;
+import org.breedinginsight.model.Program;
+import org.breedinginsight.model.ProgramLocation;
+import org.breedinginsight.model.User;
+import org.breedinginsight.services.exceptions.MissingRequiredInfoException;
+import org.breedinginsight.services.exceptions.UnprocessableEntityException;
+import org.breedinginsight.utilities.Utilities;
+import org.jooq.DSLContext;
+import tech.tablesaw.api.Table;
+import org.breedinginsight.model.Trait;
+import tech.tablesaw.columns.Column;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.math.BigInteger;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.breedinginsight.brapps.importer.services.processors.experiment.ExperimentUtilities.*;
+
+@Singleton
+@Slf4j
+public class PopulateNewPendingImportObjectsStep {
+
+    private final ExperimentSeasonService experimentSeasonService;
+    private final BrAPIObservationUnitDAO brAPIObservationUnitDAO;
+    private final DSLContext dsl;
+    private final Gson gson;
+
+    @Property(name = "brapi.server.reference-source")
+    private String BRAPI_REFERENCE_SOURCE;
+
+    @Inject
+    public PopulateNewPendingImportObjectsStep(ExperimentSeasonService experimentSeasonService,
+                                               BrAPIObservationUnitDAO brAPIObservationUnitDAO,
+                                               DSLContext dsl) {
+        this.experimentSeasonService = experimentSeasonService;
+        this.brAPIObservationUnitDAO = brAPIObservationUnitDAO;
+        this.dsl = dsl;
+        this.gson = new JSON().getGson();
+    }
+
+    public ProcessedData process(ProcessContext context, ProcessedPhenotypeData phenotypeData)
+            throws MissingRequiredInfoException, UnprocessableEntityException, ApiException {
+
+        Table data = context.getImportContext().getData();
+        ImportUpload upload = context.getImportContext().getUpload();
+        ImportContext importContext = context.getImportContext();
+
+        populatePendingImportObjects(context, phenotypeData);
+
+
+        // TODO: implement
+        return new ProcessedData();
+    }
+
```

Comment:
Is the TODO to return data instead of mutating in-place? Or something else?

Reply:
Ya, I'll clean up the interface and add some comments so it's clear what's actually happening

Reply:
Pushed changes